File: SnipeeMac/SnipeeMac/SnipeeMac/Models/Member.swift

//
//  Member.swift
//  SnipeeMac
//

import Foundation

enum MemberRole: String, Codable {
    case general = "一般"
    case admin = "管理者"
    case superAdmin = "最高管理者"
}

struct Member: Codable {
    var name: String
    var email: String
    var departments: [String]
    var role: MemberRole
    
    init(name: String, email: String, departments: [String], role: MemberRole = .general) {
        self.name = name
        self.email = email
        self.departments = departments
        self.role = role
    }
    
    var isAdmin: Bool {
        role == .admin || role == .superAdmin
    }
    
    var isSuperAdmin: Bool {
        role == .superAdmin
    }
}

File: app/models/member.js
const MemberRole = {
  GENERAL: '一般',
  ADMIN: '管理者',
  SUPER_ADMIN: '最高管理者'
};

class Member {
  constructor({
    name = '',
    email = '',
    departments = [],
    role = MemberRole.GENERAL
  } = {}) {
    this.name = name;
    this.email = email;
    this.departments = departments;
    this.role = role;
  }

  get isAdmin() {
    return this.role === MemberRole.ADMIN || this.role === MemberRole.SUPER_ADMIN;
  }

  get isSuperAdmin() {
    return this.role === MemberRole.SUPER_ADMIN;
  }
}

module.exports = { Member, MemberRole };